### DISCUSSION TITLE
PE export name length limit

### DISCUSSION THREAD

**Discussion by brianmoose** [lief-project/LIEF#821] [Q&A] [answered]
We have a DLL that has an export name that is around 3500 in length (a function that takes a bunch of function pointers, so the full signature is rather long).
Currently we parse the DLL in python with `lief.PE.parser("mylib.dll")` and it gives a warning that this symbol `is not a valid export name`.
Looking at the [source](https://github.com/lief-project/LIEF/blob/784cef9f63bb3d04e0b077dfe0add93d29e4ab78/src/PE/Parser.cpp#L939), it seems `static constexpr size_t MAX_EXPORT_NAME_SIZE = 3000; // Because of C++ mangling`
We haven't noticed any problems and other places suggest the limit is 4096.
Is 3000 a real limit and something we should be concerned about? Or does the C++ mangling comment suggest it is more of a "working limit" to take in account the mangling?

**Comment by romainthomas** [comment, marked as answer]
Sorry I missed this message. `3000` is an empirical value, so yes, we can move to 4096.
